exporter/elasticsearchexporter/factory.go

```diff
@@ -73,10 +84,49 @@ func createDefaultConfig() component.Config {
 	}
 }
 
-// createLogsExporter creates a new exporter for logs.
+func makeMarshalUnmarshalRequestFuncs(bulkIndexer *bulkIndexerPool) (
```

Comment:
Sorry, maybe I'm being dense, but I don't understand why we would want to introduce our own implementations of Request, and batching functions, vs. using the default pdata-based implementations introduced by https://github.com/open-telemetry/opentelemetry-collector/pull/9738

Reply:
I suppose one good reason for having our own would be to support a future bytes-based threshold, where the threshold should ideally be based on the serialised Elasticsearch JSON. OTOH I think it should probably also be based on the _compressed_ bytes, so that still wouldn't be a complete solution.

Reply:
> Sorry, maybe I'm being dense, but I don't understand why we would want to introduce our own implementations of Request

After perusing the code again I agree with this. It will also simplify the code in the exporter by a good degree.

Reply:
> but I don't understand why we would want to introduce our own implementations of Request, and batching functions, vs. using the default pdata-based implementations

The existing way persistent queue works now is exactly using the default pdata-based implementations along with their default marshal and unmarshal function, and only do the translations (from pdata to ES docs) in Export. We can do the same here, and yes it will simply the PR.

The question is, with this design and batchsender, we will have to define the min_items and max_items. With our custom request struct (in this PR at the moment), it will be measured in number of ES documents for both traces and logs. If we were to move to pdata, it will be measured in spans, metric data points, and log records. This will have implications on batching efficiency and will be less straightforward to find a good batching min_items config, and even harder if a good config is different for logs, metrics, and traces.

Reply:
At least for log records and spans, they will have 1:1 correspondence to ES docs, so min_items and max_items should be the same as what we have if we use a custom Request struct. For data points, it _may_ be different, but since we don't have metrics support in elasticsearch exporter now, it is a problem for later. I still think there's a way to wrap the pdata metric struct and return a different ItemsCount if we have to.

I'll make the changes to use pdata and see if it works well.